### PR TITLE
zephyr: get CDC ACM UART device from devicetree

### PR DESCRIPTION
**Pull request by jfischer-no**
Adapt to Zephyr OS changes to get CDC ACM UART device.
Remove RECOVERY_UART_DEV_NAME Kconfig option and
use DEVICE_DT_GET() in serial_adapter.c

https://github.com/zephyrproject-rtos/zephyr/pull/37085

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]

File: boot/zephyr/serial_adapter.c [hunk 1]
```diff
@@ -192,9 +192,9 @@ static int
 boot_uart_fifo_init(void)
 {
 #ifdef CONFIG_BOOT_SERIAL_UART
-	uart_dev = device_get_binding(CONFIG_RECOVERY_UART_DEV_NAME);
+	uart_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_console));
 #elif CONFIG_BOOT_SERIAL_CDC_ACM
-	uart_dev = device_get_binding(CONFIG_USB_CDC_ACM_DEVICE_NAME "_0");
+	uart_dev = DEVICE_DT_GET_ONE(zephyr_cdc_acm_uart);
```
Comment (de-nordic) [inline boot/zephyr/serial_adapter.c:197]: Two questions:
 1) shouldn't the `if (uart_dev)` that follows the line be universal for both the `zephyr_console` and `zephyr_cdc_acm_uart`?
 2) is that if even needed? wouldn't the `uart_dev` assignment fail, at compile time,  if `DEVICE_DT_GET_ONE` could not be resolved?

Comment (jfischer-no) [reply boot/zephyr/serial_adapter.c:197]: `if (uart_dev)` should be `if (!device_is_ready(uart_dev))`, updated

**Review by de-nordic — APPROVED** [review]
Looks OK.
Tested UART on nrf52840dk_nrf52840.

**Comment by nvlsianpu** [comment]
I see compilation error for nrf52dk_nrf52832 with this:
``` Console
-- Found devicetree overlay: /home/anpu/codedev/zephyrproject/bootloader/mcuboot/boot/zephyr/dts.overlay
Error: nrf52dk_nrf52832.dts.pre.tmp:559.1-13 Label or path zephyr_udc0 not found
FATAL ERROR: Syntax error parsing input tree
```

**Pull request merged**